pkg/aws/gather.go

```diff
@@ -0,0 +1,57 @@
+package aws
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/red-hat-storage/ocs-osd-deployer/utils"
+)
+
+const (
+	// https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instancedata-data-retrieval.html
+	IMDSv1Server      = "http://169.254.169.254"
+	DataConfigMapName = "aws-data"
+	CidrKey           = "vpc-cidr"
+)
+
+func GatherData(imdsServerAddr string, log logr.Logger) (map[string]string, error) {
+	var data map[string]string = map[string]string{}
+
+	// A mac address for one of the interfaces on the instance is needed
+	// to query for the VPC CIDR block.
+	log.Info("Determining mac address on instance")
+	var mac string
+	err := utils.Retry(3, time.Second, func() error {
```

Comment:
- even though it's verbose, having `1 * time.Second` shows intent more accurately 🤔